src/app/components/gv-alert/gv-alert.component.ts

```diff
@@ -0,0 +1,316 @@
+      ].concat(HttpHelpers.httpStatus);
+      this.durations = [1, 5, 10, 15, 20, 25, 30, 35, 40, 45, 50, 55];
+      this.timeUnits = ['S', 'MINUTES', 'HOURS'];
+      this.translateService
+        .get([
+          i18n('application.alerts.type.status'),
+          i18n('application.alerts.type.response_time'),
+          i18n('application.alerts.timeUnits.seconds'),
+          i18n('application.alerts.timeUnits.minutes'),
+          i18n('application.alerts.timeUnits.hours'),
+        ])
+        .toPromise()
+        .then((translations) => {
+          this.statusLabel = '' + Object.values(translations)[0];
+          this.responseTimeLabel = '' + Object.values(translations)[1];
+          this.types = [
+            {
+              label: Object.values(translations)[0],
+              value: 'STATUS',
+            },
+            {
+              label: Object.values(translations)[1],
+              value: 'RESPONSE_TIME',
+            },
+          ];
+
+          this.timeUnits = [
+            {
+              label: Object.values(translations)[2],
+              value: 'SECONDS',
+            },
+            {
+              label: Object.values(translations)[3],
+              value: 'MINUTES',
+            },
+            {
+              label: Object.values(translations)[4],
+              value: 'HOURS',
+            },
+          ];
+        });
+    }
+  }
+
+  addAlert() {
+    const statusOrReponseTimeBody = this.isStatusAlert
+      ? {
+          status_code: this.alertForm.controls.status.value,
+          status_percent: this.alertForm.controls.threshold.value,
+        }
+      : {
+          response_time: this.alertForm.controls.responseTime.value,
+        };
+    this.applicationService
+      .createApplicationAlert({
+        applicationId: this.application.id,
+        alertInput: {
+          ...statusOrReponseTimeBody,
+          enabled: true,
+          type: this.alertForm.controls.type.value,
+          duration: this.alertForm.controls.duration.value,
+          time_unit: this.alertForm.controls.timeUnit.value,
+        },
+      })
+      .toPromise()
+      .then((created) => {
+        this.notificationService.success(i18n('application.alerts.add.success'));
+        this.resetAddAlert();
+        this.resetAddAlertStatus();
+        this.alertCreated.emit(created);
+      });
+  }
+
+  onTypeChange($event: string) {
+    if ($event === 'STATUS') {
+      this.resetAddAlertStatus();
+    } else {
+      this.resetAddAlertResponseTime();
+    }
+  }
+
+  getHttpStatusLabel(value: string): string {
+    return this.httpStatus.find((http) => http.value === '' + value)?.label;
+  }
+
+  reset() {
+    this.resetAddAlert();
+    this.resetAddAlertStatus();
+  }
+
+  resetAddAlert() {
+    this.alertForm = new FormGroup({
+      type: new FormControl('STATUS', [Validators.required]),
+      duration: new FormControl('1'),
+      timeUnit: new FormControl('MINUTES', [Validators.required]),
```

Comment:
Why not using the enum generated from the openAPI descriptor ?
See `projects/portal-webclient-sdk/src/lib/model/alertTimeUnit.ts`
and `projects/portal-webclient-sdk/src/lib/model/alertType.ts`